modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/extension/v1_0/ObjectRelationshipExtensionProvider.java

```diff
@@ -232,13 +233,39 @@ public void setExtendedProperties(
 				objectRelationshipElementsParser.parse(
 					objectRelationship, entry.getValue());
 
-			if (!nestedObjectEntries.isEmpty()) {
-				DefaultObjectEntryManager defaultObjectEntryManager =
-					DefaultObjectEntryManagerProvider.provide(
-						_objectEntryManagerRegistry.getObjectEntryManager(
-							objectDefinition.getStorageType()));
-
+			DefaultObjectEntryManager defaultObjectEntryManager =
+				DefaultObjectEntryManagerProvider.provide(
+					_objectEntryManagerRegistry.getObjectEntryManager(
+						objectDefinition.getStorageType()));
+
+			if (_isManyToOneObjectRelationship(
+					objectDefinition, objectRelationship,
+					relatedObjectDefinition)) {
+
+				ObjectEntry objectEntry =
+					defaultObjectEntryManager.fetchRelatedManyToOneObjectEntry(
+						_getDefaultDTOConverterContext(
+							objectDefinition, getPrimaryKey(entity), null),
+						objectDefinition, getPrimaryKey(entity),
+						objectRelationship.getName());
+
+				if (objectEntry != null) {
+					ObjectRelatedModelsProvider objectRelatedModelsProvider =
+						_objectRelatedModelsProviderRegistry.
+							getObjectRelatedModelsProvider(
+								objectDefinition.getClassName(),
+								objectDefinition.getCompanyId(),
+								objectRelationship.getType());
+
+					objectRelatedModelsProvider.disassociateRelatedModels(
```

Comment:
Why don't we use the `defaultObjectEntryManager.disassociateRelatedModels` method instead?